### ISSUE TITLE
Allow MDX theming

### ISSUE THREAD

**Issue by fcisio** [storybookjs/storybook#12817]
**Is your feature request related to a problem? Please describe**
Styles applied in `preview.js` are applied in the canvas which is perfect, but they're also applied to the MDX docs.
There is no way to style those separately. As a result, I'm losing all default styling of the docs.

**Describe the solution you'd like**
It would be nice to separate the styles applied to the canvas from the styles applied to the docs.
This would allow for fewer side effects from the global styles and normalizing styles.

Once this is done, it would also be nice to have a way to use MDX themes. This would ensure that styles are consistent and flexible.

**Describe alternatives you've considered**
An alternative could be to expose isolated classes to the Markdown generated tags, for example `<h2 class="sbdocs">`.
With these classes exposed, it would be possible to set custom styles that would only apply to the right tags.

**Are you able to assist to bring the feature to reality?**
no

**Comment by shilman** [comment]
have you read the theming doc?

https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/theming.md

**Comment by fcisio** [comment]
> have you read the theming doc?
> 
> https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/theming.md

Hi @shilman 
I did see the doc, following the info in there I can manage to style the docs, but it's quite a bit of work. From my understanding, the current docs theme parameter only applies colors.

There's a lot of playing around with the css selectors to prevent collisions with the canvas and other docs blocks.

For example:
```css
  .sbdocs-content > p > code,
  .sbdocs-content > h1 > code,
  .sbdocs-content > h2 > code,
  .sbdocs-content > h3 > code,
  .sbdocs-content > h4 > code {
    font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, monospace;
    font-size: 85%;
    background-color: #f2f2f2;
    padding: 0.2em 0.4em;
    border-radius: 6px;
  }
```
It would be nice to be able to use themes [like these](http://jasonm23.github.io/markdown-css-themes/), we're the css selectors can be standard.

**Comment by valentinpalkovic** [comment]
Thanks for the suggestion. MDX was migrated from MDX1 to MDX3 in Storybook 8 (Canvas in Docs renders in iframes too), and you can customise MDX rendering via `MDXProvider` / custom block components. With 0 reactions in 5 years and the underlying pipeline rewritten, I'm closing as stale.
